tests/controller/modifyselection.js

```diff
@@ -4,18 +4,26 @@
  */
 
 import Document from 'ckeditor5/engine/model/document.js';
+import DataController from 'ckeditor5/engine/controller/datacontroller.js';
 import Selection from 'ckeditor5/engine/model/selection.js';
 import modifySelection from 'ckeditor5/engine/controller/modifyselection.js';
 import { setData, stringify } from 'ckeditor5/engine/dev-utils/model.js';
 
 describe( 'DataController', () => {
-	let document;
+	let document, dataController;
 
 	beforeEach( () => {
 		document = new Document();
+		dataController = new DataController( document );
+		document.schema.registerItem( 'obj' );
+		document.schema.allow( { name: 'obj', inside: '$root' } );
+		document.schema.objects.add( 'obj' );
+		document.schema.registerItem( 'inlineObj', '$inline' );
+		document.schema.objects.add( 'inlineObj' );
 		document.schema.registerItem( 'p', '$block' );
 		document.schema.registerItem( 'x', '$block' );
 		document.schema.registerItem( 'img', '$inline' );
+		document.schema.registerItem( 'b', '$inline' );
```

Comment:
This is confusing. We don't have inline containers in the model. `<b>` should rather be an attribute. This test:

```html
<p>foo<b>[]</b><inlineObj>bar</inlineObj></p>
```

have no sense. In the paragraph you can either have text or inline objects.